benchmark/kvbench/test/inference_workload_matgen.py

```
@@ -206,42 +213,85 @@ def gen_matrices_and_compute_time(
+        assert len(prefill_workers) >= len(
+            decode_workers
+        ), f"Prefill workers ({len(prefill_workers)}) must be >= decode workers ({len(decode_workers)})"
+        assert (
+            len(prefill_workers) % len(decode_workers) == 0
+        ), f"Prefill workers ({len(prefill_workers)}) must be divisible by decode workers ({len(decode_workers)})"
 
-    workers_coupling = list(zip(prefill_workers, decode_workers))
+        # Assertions
+        all_ranks = list(
+            r for worker in prefill_workers + decode_workers for r in worker
+        )
```

Ruff (0.15.15)

[warning] Unnecessary generator (rewrite as a list comprehension)

Rewrite as a list comprehension

(C400)